rewrite-java/src/main/java/org/openrewrite/java/cleanup/RenameLocalVariablesToCamelCase.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * The first character is set to lower case and existing capital letters are preserved.
+ * Special characters that are allowed in java field names `$` and `_` are removed.
+ * If a special character is removed the next valid alpha-numeric will be capitalized.
+ *
+ * Currently, unsupported:
+ *  - The recipe will not rename variables declared in a class.
+ *  - The recipe will not rename variables if the result already exists in a class or the result will be a java reserved keyword.
+ */
+public class RenameLocalVariablesToCamelCase extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Rename local variables to camel case.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Renames local variables and method parameters to java camel case naming convention.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RenameNonCompliantNames();
+    }
+
+    private static class RenameNonCompliantNames extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.VariableDeclarations.NamedVariable visitVariable(J.VariableDeclarations.NamedVariable variable, ExecutionContext ctx) {
+            Cursor parentScope = getCursorToParentScope(getCursor());
+
+            // Does not currently support renaming fields in a J.ClassDeclaration.
+            if (!(parentScope.getParent() != null && parentScope.getParent().getValue() instanceof J.ClassDeclaration)) {
+                // Does not apply to loop counters.
+                if (!(parentScope.getValue() instanceof J.ForLoop.Control)) {
+                    // Does not apply to one-character catch variables.
+                    if (!((parentScope.getValue() instanceof J.Try.Catch || parentScope.getValue() instanceof J.MultiCatch) &&
+                            variable.getSimpleName().length() == 1)) {
+                        // Regex to match java naming convention.
+                        Pattern pattern = Pattern.compile("^[a-z][a-zA-Z0-9]*$");
+                        if (!pattern.matcher(variable.getSimpleName()).matches()) {
+                            String toName = convertToCamelCase(variable.getSimpleName());
+                            // An identifier does not exit in the class.
+                            if (!toNameFound(toName, getCursor(), ctx)) {
+                                doAfterVisit(new RenameVariable<>(variable, toName));
+                                return variable;
+                            }
+                        }
+                    }
+                }
+            }
+            return super.visitVariable(variable, ctx);
+        }
+    }
+
+    private static boolean toNameFound(String toName, Cursor cursor, ExecutionContext ctx) {
+        new FindName(toName).visit(cursor.firstEnclosing(J.CompilationUnit.class), ctx);
```

Comment:
Have you considered adding a static `find` method to `FindName` similar to `FindTypes.find`?  Then adding values to the `ExecutionContext` would not be necessary.

Reply:
No, that's a great idea - refactored the method to avoid using the `ExecutionContext`. Thanks!